### PR TITLE
Use /self/proc/maps to determine root

### PR DESCRIPTION
**Pull request by mrjackv**

### CODE REVIEW

File: lib/Support/PathList.cpp [hunk 1]
```diff
@@ -24,6 +27,32 @@ std::string getCurrentExecutableFullPath() {
   return FullPath.str().str();
 }
 
+std::string getCurrentRoot() {
+  using llvm::sys::path::parent_path;
+
+  std::ifstream Maps("/proc/self/maps");
```
Comment (aleclearmind) [inline lib/Support/PathList.cpp:33]: Let's avoid `ifstream`, let's use `MemoryBuffer::getFile` and then `StringRef.split("\n")`.

File: lib/Support/PathList.cpp [hunk 2]
```diff
@@ -24,6 +27,32 @@ std::string getCurrentExecutableFullPath() {
   return FullPath.str().str();
 }
 
+std::string getCurrentRoot() {
```
Comment (aleclearmind) [inline lib/Support/PathList.cpp:30]: This function sometimes returns `""`, but callers do not handle such situation.
Let's abort in those case so we can guarantee that if the function returns, returns sensible data..

**Review by aleclearmind — CHANGES_REQUESTED** [review]

File: lib/Support/PathList.cpp (hunk 1 above)
Comment (mrjackv) [reply lib/Support/PathList.cpp]: Done

File: lib/Support/PathList.cpp (hunk 2 above)
Comment (mrjackv) [reply lib/Support/PathList.cpp:26]: Fixed

File: lib/Support/PathList.cpp [hunk 3]
```diff
@@ -2,28 +2,56 @@
...
+
+  auto MaybeBuffer = llvm::MemoryBuffer::getFileAsStream("/proc/self/maps");
+  if (!MaybeBuffer)
+    revng_abort("Unable to open /proc/self/maps");
+
+  auto Maps = (*MaybeBuffer)->getBuffer();
+  llvm::SmallVector<llvm::StringRef, 128> MapsLines;
+  Maps.split(MapsLines, "\n");
+
+  for (auto Line = MapsLines.begin(); Line < MapsLines.end(); Line++) {
```
Comment (aleclearmind) [inline lib/Support/PathList.cpp:37]: Can you?
```suggestion
  for (const auto &Line : MapsLines) {
```

**Comment by aleclearmind** [comment]
Merged in 13bf5a7. Thanks!

Please remember to put the dot the end of lines in commits `:)`

**Pull request closed without merging**